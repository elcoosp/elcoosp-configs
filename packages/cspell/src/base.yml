dictionaryDefinitions:
  - name: ts-libs
    path: ./ts-libs.txt
    addWords: true
dictionaries:
  - ts-libs
ignorePaths:
  - node_modules
  - pnpm-lock.yaml
  - ./ts-libs.txt
