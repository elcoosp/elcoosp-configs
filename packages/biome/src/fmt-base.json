{
  "organizeImports": {
    "enabled": true
  },
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "formatter": {
    "indentStyle": "space",
    "ignore": ["__snapshots__"]
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single",
      "semicolons": "asNeeded"
    }
  },
  "css": {
    "parser": {
      "cssModules": true
    }
  },
  "linter": {
    "enabled": false
  }
}
