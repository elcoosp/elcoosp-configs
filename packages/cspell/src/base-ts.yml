dictionaryDefinitions:
  - name: ts-libs
    path: ./ts-libs.txt
    addWords: true
dictionaries:
  - ts-libs
  - typescript
  - node
  - npm
ignorePaths:
  - node_modules
  - pnpm-lock.yaml
  - ./ts-libs.txt
