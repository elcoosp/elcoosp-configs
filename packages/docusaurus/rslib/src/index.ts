import { type RslibConfig, defineConfig } from "@rslib/core";
export type Config = {
	preset: "dual";
};
const presets = {
	dual: {
		lib: [
			{
				format: "esm",
				syntax: "es2021",
				dts: true,
			},
			{
				format: "cjs",
				output: {
					distPath: {
						root: "./dist/cjs",
					},
				},
			},
		],
	},
} as const satisfies Record<Config["preset"], RslibConfig>;
export function createRsLibConfig({ preset }: Config) {
	return defineConfig(presets[preset]);
}
