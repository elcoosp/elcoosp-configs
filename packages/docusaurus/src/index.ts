import type * as Preset from '@docusaurus/preset-classic';
import type { Config } from '@docusaurus/types';
import { deepmerge } from 'deepmerge-ts';
import { themes as prismThemes } from 'prism-react-renderer';
import { titleCase } from 'title-case';

type DerivedConfig = {
  githubUrl: string;
} & Pick<Config, 'organizationName' | 'title'> &
  StaticConfig;
type StaticConfig = {
  projectName: string;
} & Pick<Config, 'organizationName'>;
export function themeConfig({
  title,
  githubUrl,
  projectName,
}: DerivedConfig): Preset.ThemeConfig {
  return {
    image: 'img/social-card.png',
    navbar: {
      title,
      logo: {
        alt: `${title} logo`,
        src: 'img/logo.svg',
      },
      items: [
        {
          type: 'docSidebar',
          sidebarId: 'docsSidebar',
          position: 'left',
          label: 'Docs',
        },
        { to: '/blog', label: 'Blog', position: 'left' },
        {
          href: githubUrl,
          label: 'GitHub',
          position: 'right',
        },
      ],
    },
    footer: {
      style: 'dark',
      links: [
        {
          title: 'Docs',
          items: [
            {
              label: 'Docs',
              to: '/docs/intro',
            },
          ],
        },
        {
          title: 'Community',
          items: [
            {
              label: 'Stack Overflow',
              href: `https://stackoverflow.com/questions/tagged/${projectName}`,
            },
          ],
        },
        {
          title: 'More',
          items: [
            {
              label: 'Blog',
              to: '/blog',
            },
            {
              label: 'GitHub',
              href: githubUrl,
            },
          ],
        },
      ],
    },
    prism: {
      theme: prismThemes.github,
      darkTheme: prismThemes.dracula,
    },
  };
}
export function classicPreset(
  config: DerivedConfig,
): ['classic', Preset.Options] {
  return [
    'classic',
    {
      docs: {
        sidebarPath: './sidebars.ts',
        editUrl: `${config.githubUrl}/tree/main/documentation/blog`,
      },
      blog: {
        showReadingTime: true,
        feedOptions: {
          type: ['rss', 'atom'],
          xslt: true,
        },
        editUrl: `${config.githubUrl}/tree/main/documentation/docs`,
        onInlineTags: 'warn',
        onInlineAuthors: 'warn',
        onUntruncatedBlogPosts: 'warn',
      },
      theme: {
        customCss: './src/css/custom.css',
      },
    } satisfies Preset.Options,
  ];
}

export function createDocusaurusConfig(
  { projectName, organizationName }: StaticConfig,
  getConfig: (derivedConfig: DerivedConfig) => Partial<Config>,
): Config {
  const githubUrl = `https://github.com/${organizationName}/${projectName}`;
  const deploy = 'vercel' as const;
  const url = { vercel: `https://${projectName}.vercel.app` }[deploy];
  const title = titleCase(projectName);
  const baseUrl = { vercel: '/' }[deploy];
  return deepmerge<[Config, Config]>(
    getConfig({
      githubUrl,
      organizationName,
      title,
      projectName,
    }) as Config,
    {
      url,
      baseUrl,
      organizationName,
      title,
      favicon: 'img/favicon.ico',
      onBrokenLinks: 'throw',
      onBrokenMarkdownLinks: 'warn',
      i18n: {
        defaultLocale: 'en',
        locales: ['en'],
      },
      future: {
        experimental_faster: true,
      },
    },
  );
}
