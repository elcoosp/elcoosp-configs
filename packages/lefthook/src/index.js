console.error(
  'There is no js files exported in this package, check the src/*.yml lefthook preset files',
);
