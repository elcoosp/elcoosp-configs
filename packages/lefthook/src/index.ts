export const ERROR_MESSAGE =
  'There is no js files exported in this package, check the src/*.yml lefthook preset files';
console.error(ERROR_MESSAGE);
