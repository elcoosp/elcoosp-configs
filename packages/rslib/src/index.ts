import type { RslibConfig } from '@rslib/core';
import { readPackage } from 'read-pkg';
export type Config = {
  preset: 'dual';
};
const presets = {
  dual: async () => {
    const pkg = await readPackage();
    // TODO check that match inside package.json
    const cjsOutDistPathRoot = './dist/cjs';
    return {
      lib: [
        {
          format: 'esm',
          syntax: 'es2021',
          dts: true,
        },
        {
          format: 'cjs',
          output: {
            distPath: {
              root: cjsOutDistPathRoot,
            },
          },
        },
      ],
    };
  },
} as const satisfies Record<Config['preset'], () => Promise<RslibConfig>>;
export async function createRsLibConfig({
  preset,
}: Config): Promise<RslibConfig> {
  // Should not call defineConfig otherwise build fail
  return await presets[preset]();
}
