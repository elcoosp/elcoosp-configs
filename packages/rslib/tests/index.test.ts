import { expect, test } from 'vitest';
import { createRsLibConfig } from '../src/index';

test('squared', async () => {
  expect(await createRsLibConfig({ preset: 'dual' })).toMatchSnapshot();
});
