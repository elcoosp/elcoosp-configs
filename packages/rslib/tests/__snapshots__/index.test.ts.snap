// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createRsLibConfig 1`] = `
{
  "lib": [
    {
      "bundle": true,
      "dts": true,
      "format": "esm",
      "syntax": "es2021",
    },
    {
      "bundle": true,
      "format": "cjs",
      "output": {
        "distPath": {
          "root": "./dist/cjs",
        },
      },
    },
  ],
}
`;
